### PR TITLE
psh: add gibber test

### PR DESCRIPTION
**Pull request by damianloew**
## Description

 - psh: add assert_unprintable() to psh test module 
 - psh: add test-gibber

## Motivation and Context

JIRA: PD-77

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: psh/test-gibber.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    psh.assert_cmd(p, string.digits, EXPECTED, msg)
+
+
+def assert_punctuation(p):
+    msg = 'Wrong output for punctuation characters'
+    psh.assert_cmd(p, string.punctuation, EXPECTED, msg)
+
+
+def assert_unprintable(p):
+    for ascii_code in range(32):
```
Comment (kemonats) [inline psh/test-gibber.py:46]: Without additional code for `DEL` character.
```suggestion
    for ascii_code in (*range(32), 127);
```
or using `itertools.chain()`
```suggestion
    import itertools
    for ascii_code in itertools.chain(range(32), range(127,128)):
```

Comment (damianloew) [reply psh/test-gibber.py]: Yeah, that looks better, thanks :)

File: psh/test-gibber.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        # LF or ^D, double prompt is expected
+        if ascii_code == 4 or ascii_code == 10:
+            psh.assert_prompt(p)
+
+
+def assert_not_ascii(p):
+    for code in range(128, 256):
+        cmd = chr(code)
+        msg = f'Wrong output for character with the following code: {code}'
+        psh.assert_prompt_after_cmd(p, cmd, msg)
```
Comment (xvuko) [inline psh/test-gibber.py:69]: Waiting for prompt after every characters takes a lot of time.

You could test all nonprintable characters without side effects at once:
```python
s = ''.join(chr(x) for x in sorted({*range(0, 32),*range(127, 256)} - {3, 4, 10, 28, 127}))
output = psh.assert_prompt_after_cmd(p, characters, msg)
assert output == '', f"Unexpected output for non-printable characters: {output!r}"
```
^ Code above assumes assert_prompt_after_cmd returnes output.

File: psh/test-gibber.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+
+def assert_digits(p):
+    msg = 'Wrong output for all digits'
+    psh.assert_cmd(p, string.digits, EXPECTED, msg)
+
+
+def assert_punctuation(p):
+    msg = 'Wrong output for punctuation characters'
+    psh.assert_cmd(p, string.punctuation, EXPECTED, msg)
```
Comment (xvuko) [inline psh/test-gibber.py:42]: Mabye do all printables in single prompt?

File: psh/test-gibber.py (hunk 2 above)
Comment (damianloew) [reply psh/test-gibber.py]: Yeah, this solution is much faster. I think it will be better.

File: psh/test-gibber.py (hunk 3 above)
Comment (damianloew) [reply psh/test-gibber.py]: Yeah, please check the latest force-push

**Review by xvuko — APPROVED** [review]

**Pull request merged**